src/test/regress/sql/pg16.sql

```diff
@@ -650,10 +622,80 @@ RESET citus.enable_ddl_propagation;
 -- by default, admin option is false, inherit is true, set is true
 GRANT role3 TO role4;
 GRANT role3 TO role5 WITH ADMIN TRUE, INHERIT FALSE, SET FALSE;
-SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option FROM pg_auth_members WHERE roleid::regrole::text = 'role3' ORDER BY 1, 2;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role3' ORDER BY 1, 2;
 
 DROP ROLE role3, role4, role5;
 
+-- Test that everything works fine for roles that are propagated
+-- Grant role with admin and inherit options set to true
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role6;
+CREATE ROLE role7;
+GRANT role6 TO role7 WITH ADMIN OPTION, INHERIT TRUE;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option FROM pg_auth_members
```

Comment:
In order to be able to test that we are propagating correctly, we should run the same command in the worker node as well to see that we get the same output as in the coordinator. You can run commands in the worker node by:
```sql
-- connect to worker node
\c - - - :worker_1_port
-- run the same command
SELECT roleid::regrole::text AS role, member::regrole::text, admin_option FROM pg_auth_members
WHERE roleid::regrole::text = 'role6' ORDER BY 1, 2;
-- reconnect to master node
\c - - - :master_port
```